mlflow/entities/metic_value_conversion_utils.py

```diff
@@ -0,0 +1,141 @@
+import sys
+
+
+def is_module_imported(module_name: str) -> bool:
+    return module_name in sys.modules
+
+
+def __converter_requires(module_name: str):
+    """Wrapper function that checks if specified `module_name` is already imported before invoking wrapped function."""
+
+    def decorator(func):
+        def wrapper(x):
+            if not is_module_imported(module_name):
+                return x
+
+            return func(x)
+
+        return wrapper
+
+    return decorator
+
+
+def convert_metric_value_to_str_if_possible(x) -> str:
+    if x is None or type(x) == str:
+        return x
+
+    converter_fns_to_try = [
+        convert_metric_value_to_str_if_pyspark_mlparam,
+        convert_metric_value_to_str_if_ndarray,
+        convert_metric_value_to_str_if_tensorflow_tensor,
+        convert_metric_value_to_str_if_torch_tensor,
+    ]
+
+    for converter_fn in converter_fns_to_try:
+        possible_str = converter_fn(x)
+        if type(possible_str) == str:
+            return possible_str
+
+    return str(x)
```

Comment:
@ntakouris I'm still not fully following why we need to special case string conversion for the use of tensors / pyspark mlparams as metric keys (names). I think it should be sufficient to convert all `key` inputs to string using `str()`. Is there a particular use case you have for using a tensor as a metric name?

Reply:
I've not pushed my local changes.